plugwise/smile.py

```diff
@@ -485,32 +491,32 @@ async def _set_schedule_state_legacy(self, name, status):
         await self._request(uri, method="put", data=data)
         return True
 
-    async def set_schedule_state(self, loc_id, name, state):
+    async def set_schedule_state(self, loc_id: str, name: str, state: str) -> bool:
         """Set the Schedule, with the given name, on the relevant Thermostat.
         Determined from - DOMAIN_OBJECTS.
         """
         if self._smile_legacy:
             return await self._set_schedule_state_legacy(name, state)
 
-        schema_rule = self._rule_ids_by_name(str(name), loc_id)
+        schema_rule = self._rule_ids_by_name(name, loc_id)
         if not schema_rule or schema_rule is None:
             return False
 
-        schema_rule_id = next(iter(schema_rule))
-        info = ""
+        schema_rule_id: str = next(iter(schema_rule))
+        info: str = ""
```

Comment:
Already inferred, not needed. 

```suggestion
        info = ""
```